로컬에서 변경사항을 github로 푸시하려고 하는데 아래와 같은 메시지가 뜨면서 거부되는 경우가 있다.  
```
To https://github.com/yunkangmin/11st_memo.git
 ! [rejected]        master -> master (fetch first)
error: failed to push some refs to 'https://github.com/yunkangmin/11st_memo.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```
해석해보자면 아래와 같이 github에 커밋이력과 로컬 커밋이력이 달라서 푸시가 거부된 것이다.
(다른 저장소에 푸시할 경우도 이런 경우가 생긴다(푸시할 때 브랜치가 main인지 master인지 먼저 확인하자).
![image](https://user-images.githubusercontent.com/33191974/141285108-ece656f3-da50-4077-a58e-845fd7cbe020.png)

해결법은 아래와 같다.
1. 먼저 로컬에서 git pull origin master(or main)  
git pull을 하면 github에 있는 커밋이력을 로컬로 가져와 병합한다.  
그리고 머지 커밋을 하게 된다.
![image](https://user-images.githubusercontent.com/33191974/141287356-c3da9038-908f-4fe7-a1ec-8062c38a917f.png)

3. 다시 push

